chapters/glossary/glossary.rst

```diff
@@ -0,0 +1,78 @@
+    Client
+        A Python client that connects to a remote Jina gateway.
+    Container
+        A Docker container image is a lightweight, standalone, executable package of software that includes everything needed to run an application: code, runtime, system tools, system libraries and settings.
+    Crafter
+        Crafters are a class of Executor whose purpose is to alter a Document to match a certain requirement. This could include tasks such as resizing images or reducing all text in a document to lowercase.
+    Document
+        Documents are the most basic data types available within Jina. Anything you want to search for within Jina is considered a document, this could include images, sounds clips or text documents.
+    Driver
+        Drivers wrap an Executor and interprets network traffic into a format the Executor can understand. All Executors must have at least one driver. A Driver is further wrapped by a Pea.
+    Embedding space
+        Embedding space is the vector space in which the data is embedded after dimensionality reduction. Depending on how the space is created, semantically similar items are put together and dissimilar items are kept far apart. Position (distance and direction) in the vector space can encode semantics in a good embedding. `reference_1 <https://link.springer.com/referenceworkentry/10.1007%2F978-0-387-73003-5_573#:~:text=Embedding%20space%20is%20the%20space,than%20of%20the%20ambient%20space.>`_
+    Encoder
+        Encoders are a class of Executor whose purpose is to take a variable-length input and output a fixed-length semantic embedding vector. This is achieved by passing the input to a pre-created language model which returns a fixed length context specific vector.
+    Evaluator
+        Evaluators are a class of Executor whose purpose is to provide advanced evaluation metrics on the performance of a search system. Distance metrics such as cosine distance and euclidean distance. As well evaluation metrics such as precision, and recall between the results and the ground truth (provided by user) can be computed.
+    Executor
+        Executors represent an algorithmic class within the Jina framework. Examples include Ranker classes, Evaluator classes etc.
+    Flow
+        A Flow is created for each high-level task in Jina, such as querying or indexing. It manages the state and context of the Pods or Peas who work together to complete this high-level task.
+    Index Mode
+        During Index mode, Jina performs collecting, parsing, and storing of the data to facilitate fast and accurate information retrieval.
+    Indexer
+        Encoders are a class of Executor whose purpose is to save and retrieve vectors and key-value information from storage.
+    JAML
+        JAML is a Jina YAML parser that supports loading, dumping and substituting variables.
```

Comment:
substituting `key-value` information

Reply:
Could you also make the suggestions on the google doc? just to make sure we all agree on the definitions?